qfieldsync/core/cloud_api.py

```diff
@@ -374,9 +541,35 @@ def set_token(self, token: str, update_auth: bool = False) -> None:
 
         self.token_changed.emit()
 
+    def is_authenticated(self) -> bool:
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            return self.has_token()
+        if self.auth_method == CloudAuthMethod.SSO:
+            # If there is no username, it means a request to get it
+            # must be done before being considered authenticated.
+            return self.auth_config and self.current_username
+        return False
```

Comment:
```suggestion
        if self.auth_method == CloudAuthMethod.NONE:
            return False
        elif self.auth_method == CloudAuthMethod.CREDENTIALS:
            return self.has_token()
        elif self.auth_method == CloudAuthMethod.SSO:
            # If there is no username, it means a request to get it
            # must be done before being considered authenticated.
            return self.auth_config and self.current_username
        else:
            raise NotImplementedError("Unknown authentication method {self.auth_method}!")
```

I think this is more future proof.